pkg/controller/deployment/deployment_overrides_test.go

```diff
@@ -396,3 +400,145 @@ func TestParseArgMap(t *testing.T) {
 		t.Fatalf("unexpected update to arg map, diff = %v", cmp.Diff(wantMap, argMap))
 	}
 }
+
+func TestWithContainerResourcesOverrideHook(t *testing.T) {
+	tests := []struct {
+		name             string
+		sourceDeployment appsv1.Deployment
+		overrideFn       overrideResourcesFunc
+		expected         appsv1.Deployment
+	}{
+		{
+			name: "override resources replaces empty container resouces",
+			sourceDeployment: appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "test-namespace",
+				},
+				Spec: appsv1.DeploymentSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name: "test-container",
+								},
+							},
+						},
+					},
+				},
+			},
+			overrideFn: func(_ certmanagerinformer.CertManagerInformer, _ string) (v1alpha1.CertManagerResourceRequirements, error) {
```

Comment:
@arkadeepsen I don't the value running `withContainerResourcesOverrideHook()` in the test if are also passing the override test function. We can ditch this in favour of covering this as part of e2e.

Reply:
This unit test is for checking whether the `withContainerResourcesOverrideHook` is working correctly or not with the inputs provided. The e2e test will anyway test the overall functionality of the feature.

Reply:
Ok. I'll remove the unit test for `withContainerResourcesOverrideHook`.